src/promptflow/tests/sdk_cli_test/unittests/test_trace.py

```diff
@@ -2,22 +2,88 @@
+from opentelemetry.sdk.trace import TracerProvider
 
-from promptflow._constants import SpanAttributeFieldName
-from promptflow._core.operation_context import OperationContext
-from promptflow._trace._start_trace import _provision_session
+from promptflow._constants import ResourceAttributeFieldName, TraceEnvironmentVariableName
+from promptflow._trace._start_trace import (
+    _create_resource,
+    _is_tracer_provider_configured,
+    _provision_session_id,
+    setup_exporter_from_environ,
+)
 
 
 @pytest.mark.sdk_test
 @pytest.mark.unittest
-class TestTrace:
-    def test_session_id_in_operation_context(self):
-        session_id = _provision_session()
-        operation_context = OperationContext.get_instance()
-        assert session_id == operation_context._get_otel_attributes()[SpanAttributeFieldName.SESSION_ID]
-
-    def test_provision_session_multiple_times(self):
-        session_id = _provision_session()
-        session_id_twice = _provision_session()
-        assert session_id == session_id_twice
+class TestStartTrace:
+    def test_create_resource(self) -> None:
+        session_id = str(uuid.uuid4())
+        resource1 = _create_resource(session_id=session_id)
+        assert resource1.attributes[ResourceAttributeFieldName.SESSION_ID] == session_id
+        assert ResourceAttributeFieldName.EXPERIMENT_NAME not in resource1.attributes
+
+        experiment = "test_experiment"
+        resource2 = _create_resource(session_id=session_id, experiment=experiment)
+        assert resource2.attributes[ResourceAttributeFieldName.SESSION_ID] == session_id
+        assert resource2.attributes[ResourceAttributeFieldName.EXPERIMENT_NAME] == experiment
+
+    def test_setup_exporter_from_environ(self) -> None:
+        assert not _is_tracer_provider_configured()
+
+        # set some required environment variables
+        endpoint = "http://localhost:23333/v1/traces"
+        session_id = str(uuid.uuid4())
+        experiment = "test_experiment"
+        with patch.dict(
+            os.environ,
+            {
+                OTEL_EXPORTER_OTLP_ENDPOINT: endpoint,
+                TraceEnvironmentVariableName.SESSION_ID: session_id,
+                TraceEnvironmentVariableName.EXPERIMENT: experiment,
+            },
+            clear=True,
+        ):
+            setup_exporter_from_environ()
+
+        assert _is_tracer_provider_configured()
+        tracer_provider: TracerProvider = trace.get_tracer_provider()
+        assert session_id == tracer_provider._resource.attributes[ResourceAttributeFieldName.SESSION_ID]
+        assert experiment == tracer_provider._resource.attributes[ResourceAttributeFieldName.EXPERIMENT_NAME]
+
+    def test_provision_session_id(self) -> None:
+        # no specified, session id should be a valid UUID
+        session_id = _provision_session_id(specified_session_id=None)
+        assert session_id == str(uuid.UUID(session_id, version=4))
```

Comment:
why this works? the two uuid should be different?

Reply:
no, you can regard this as a validation for string that is a valid UUID